lib/qless/worker/base.rb

```diff
@@ -23,18 +23,21 @@ def initialize(reserver, options = {})
         @options = options
 
         # Our logger
-        @output = options.fetch(:output) { $stdout }
-        @log = Logger.new(output)
-        @log_level = options[:log_level] || Logger::WARN
-        @log.level = @log_level
-        @log.formatter = options.fetch(:log_formatter) {
-          proc do |severity, datetime, progname, msg|
-            "#{datetime}: #{msg}\n"
+        @log = options.fetch(:logger) do
+          @output = options.fetch(:output, $stdout)
+          Logger.new(output).tap do |logger|
+            @log_level = options.fetch(:log_level, Logger::WARN)
+            logger.level = @log_level
+            logger.formatter = options.fetch(:log_formatter) {
```

Comment:
We might consider turning this into a `do` `end` style block, instead of the curly braces, because it's multi-line, but I know that sometimes @benkirzhner likes curly braces over do/ends.